### PR TITLE
JSON/API Lesson Cleanup

### LINKED ISSUE #3962: Make Get JSON with the jQuery getJSON Method challenge quote agnostic
Challenge [Waypoint: Get JSON with the jQuery getJSON Method](http://freecodecamp.com/challenges/waypoint-get-json-with-the-jquery-getjson-method#?solution=fccss%0A%20%24(document).ready(function()%20%7B%0A%20%20%20%0A%20%20%20%24(%22%23getMessage%22).on(%22click%22%2C%20function()%7B%0A%20%20%20%20%2F%2F%20Only%20change%20code%20below%20this%20line.%0A%0A%20%20%20%20%24.getJSON(%22%2Fjson%2Fcats.json%3Fcallback%3D%22%2C%20function(json)%20%7B%0A%20%20%20%20%20%20%24('.message').html(JSON.stringify(json))%3B%0A%20%20%20%20%7D)%3B%0A%0A%20%20%20%20%2F%2F%20Only%20change%20code%20above%20this%20line.%0A%20%20%20%7D)%3B%0A%20%20%20%0A%20%7D)%3B%0Afcces%0A%0A%3Cdiv%20class%3D%22container-fluid%22%3E%0A%20%3Cdiv%20class%20%3D%20%22row%20text-center%22%3E%0A%20%20%20%3Ch2%3ECat%20Photo%20Finder%3C%2Fh2%3E%0A%20%3C%2Fdiv%3E%0A%20%3Cdiv%20class%20%3D%20%22row%20text-center%22%3E%0A%20%20%20%3Cdiv%20class%20%3D%20%22col-xs-12%20well%20message%22%3E%0A%20%20%20%20%20The%20message%20will%20go%20here%0A%20%20%20%3C%2Fdiv%3E%0A%20%3C%2Fdiv%3E%0A%20%3Cdiv%20class%20%3D%20%22row%20text-center%22%3E%0A%20%20%20%3Cdiv%20class%20%3D%20%22col-xs-12%22%3E%0A%20%20%20%20%20%3Cbutton%20id%20%3D%20%22getMessage%22%20class%20%3D%20%22btn%20btn-primary%22%3E%0A%20%20%20%20%20%20%20Get%20Message%0A%20%20%20%20%20%3C%2Fbutton%3E%0A%20%20%20%3C%2Fdiv%3E%0A%20%3C%2Fdiv%3E%0A%3C%2Fdiv%3E%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/46.0.2490.80 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

My code:
```html
<script>
 $(document).ready(function() {
   
   $("#getMessage").on("click", function(){
    // Only change code below this line.

    $.getJSON("/json/cats.json?callback=", function(json) {
      $('.message').html(JSON.stringify(json));
    });

    // Only change code above this line.
   });
   
 });
</script>

<div class="container-fluid">
 <div class = "row text-center">
   <h2>Cat Photo Finder</h2>
 </div>
 <div class = "row text-center">
   <div class = "col-xs-12 well message">
     The message will go here
   </div>
 </div>
 <div class = "row text-center">
   <div class = "col-xs-12">
     <button id = "getMessage" class = "btn btn-primary">
       Get Message
     </button>
   </div>
 </div>
</div>

```

### LINKED ISSUE #3969: Fix Get JSON challenge text & quote specificity
Challenge [Waypoint: Get JSON with the jQuery getJSON Method](http://www.freecodecamp.com/challenges/waypoint-get-json-with-the-jquery-getjson-method#?solution=fccss%0A%20%24(document).ready(function()%20%7B%0A%20%20%20%0A%20%20%20%24(%22%23getMessage%22).on(%22click%22%2C%20function()%7B%0A%20%20%20%20%2F%2F%20Only%20change%20code%20below%20this%20line.%0A%24.getJSON(%22%2Fjson%2Fcats.json%3Fcallback%3D%22%2C%20function%20(json)%7B%0A%20%20%0A%20%20%24(%22.message%22).html(JSON.stringify(json))%0A%7D)%3B%0A%0A%0A%20%20%20%20%2F%2F%20Only%20change%20code%20above%20this%20line.%0A%20%20%20%7D)%3B%0A%20%20%20%0A%20%7D)%3B%0Afcces%0A%0A%3Cdiv%20class%3D%22container-fluid%22%3E%0A%20%3Cdiv%20class%20%3D%20%22row%20text-center%22%3E%0A%20%20%20%3Ch2%3ECat%20Photo%20Finder%3C%2Fh2%3E%0A%20%3C%2Fdiv%3E%0A%20%3Cdiv%20class%20%3D%20%22row%20text-center%22%3E%0A%20%20%20%3Cdiv%20class%20%3D%20%22col-xs-12%20well%20message%22%3E%0A%20%20%20%20%20The%20message%20will%20go%20here%0A%20%20%20%3C%2Fdiv%3E%0A%20%3C%2Fdiv%3E%0A%20%3Cdiv%20class%20%3D%20%22row%20text-center%22%3E%0A%20%20%20%3Cdiv%20class%20%3D%20%22col-xs-12%22%3E%0A%20%20%20%20%20%3Cbutton%20id%20%3D%20%22getMessage%22%20class%20%3D%20%22btn%20btn-primary%22%3E%0A%20%20%20%20%20%20%20Get%20Message%0A%20%20%20%20%20%3C%2Fbutton%3E%0A%20%20%20%3C%2Fdiv%3E%0A%20%3C%2Fdiv%3E%0A%3C%2Fdiv%3E%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

Exercise not accepting single quotes & has "You should have at least on closing set of brackets and parenthesis." instead of "You should have at least one closing set of brackets and parenthesis." 

My code:
```html
<script>
 $(document).ready(function() {
   
   $("#getMessage").on("click", function(){
    // Only change code below this line.
$.getJSON('/json/cats.json?callback=', function (json){
  
  $('.message').html(JSON.stringify(json))
});


    // Only change code above this line.
   });
   
 });
</script>

<div class="container-fluid">
 <div class = "row text-center">
   <h2>Cat Photo Finder</h2>
 </div>
 <div class = "row text-center">
   <div class = "col-xs-12 well message">
     The message will go here
   </div>
 </div>
 <div class = "row text-center">
   <div class = "col-xs-12">
     <button id = "getMessage" class = "btn btn-primary">
       Get Message
     </button>
   </div>
 </div>
</div>

```

### LINKED ISSUE #3972: remove `?callback=` from Get JSON challenge
Challenge [Waypoint: Get JSON with the jQuery getJSON Method](http://freecodecamp.com/challenges/waypoint-get-json-with-the-jquery-getjson-method) has an issue.

It has no purpose, especially considering this is a simple GET request and not a ajax jsonp request.

### LINKED ISSUE #4010: Spaces needed
There needs to be a space after Latitude, and one before and after Longitude, and maybe a comma as well - $("#data").html("Latitiude " + position.coords.latitude + ", Longitude " + position.coords.longitude);

Turns
latitude-1.2677468longitude-1.2677568
into
Latitude -1.2677468, Longitude -1.2677568

### LINKED ISSUE #4016: Help, it is not rendering the cat pics :)
Challenge [Waypoint: Convert JSON Data to HTML](http://freecodecamp.com/challenges/waypoint-convert-json-data-to-html#?solution=fccss%0A%20%20%24(document).ready(function()%20%7B%0A%0A%20%20%20%20%24(%22%23getMessage%22).on(%22click%22%2C%20function()%20%7B%0A%E2%80%89%E2%80%89%20%20%20%20%24.getJSON(%22%2Fjson%2Fcats.json%3Fcallback%3D%22%2C%20function(%20json%20)%20%7B%0A%0A%20%20%20%20%20%20%20%20%2F%2F%20Only%20change%20code%20below%20this%20line.%0A%0A%0Ajson.map(function(val)%20%7B%0A%0A%E2%80%89%E2%80%89html%20%3D%20html%20%2B%20%22%3Cdiv%20class%20%3D%20'cat'%3E%22%0A%0A%E2%80%89%E2%80%89html%20%3D%20html%20%2B%20'%3Cdiv%3E'%20%2B%20val%20%2B%20'%3C%2Fdiv%3E'%3B%0A%0A%E2%80%89%E2%80%89html%20%3D%20html%20%2B%20%22%3C%2Fdiv%3E%3Cbr%2F%3E%22%0A%0A%7D)%3B%0A%0A%0A%0A%0A%0A%0A%20%20%20%20%20%20%20%20%2F%2F%20Only%20change%20code%20above%20this%20line.%0A%0A%E2%80%89%E2%80%89%E2%80%89%E2%80%89%20%20%20%20%24(%22.message%22).html(html)%3B%0A%0A%E2%80%89%E2%80%89%20%20%20%20%7D)%3B%0A%20%20%20%20%7D)%3B%0A%20%20%7D)%3B%0Afcces%0A%0A%3Cdiv%20class%3D%22container-fluid%22%3E%0A%20%3Cdiv%20class%20%3D%20%22row%20text-center%22%3E%0A%20%20%20%3Ch2%3ECat%20Photo%20Finder%3C%2Fh2%3E%0A%20%3C%2Fdiv%3E%0A%20%3Cdiv%20class%20%3D%20%22row%20text-center%22%3E%0A%20%20%20%3Cdiv%20class%20%3D%20%22col-xs-12%20well%20message%22%3E%0A%20%20%20%20%20The%20message%20will%20go%20here%0A%20%20%20%3C%2Fdiv%3E%0A%20%3C%2Fdiv%3E%0A%20%3Cdiv%20class%20%3D%20%22row%20text-center%22%3E%0A%20%20%20%3Cdiv%20class%20%3D%20%22col-xs-12%22%3E%0A%20%20%20%20%20%3Cbutton%20id%20%3D%20%22getMessage%22%20class%20%3D%20%22btn%20btn-primary%22%3E%0A%20%20%20%20%20%20%20Get%20Message%0A%20%20%20%20%20%3C%2Fbutton%3E%0A%20%20%20%3C%2Fdiv%3E%0A%20%3C%2Fdiv%3E%0A%3C%2Fdiv%3E%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/46.0.2490.80 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

My code:
```html
<script>
  $(document).ready(function() {

    $("#getMessage").on("click", function() {
      $.getJSON("/json/cats.json?callback=", function( json ) {

        // Only change code below this line.


json.map(function(val) {

  html = html + "<div class = 'cat'>"

  html = html + '<div>' + val + '</div>';

  html = html + "</div><br/>"

});






        // Only change code above this line.

        $(".message").html(html);

      });
    });
  });
</script>

<div class="container-fluid">
 <div class = "row text-center">
   <h2>Cat Photo Finder</h2>
 </div>
 <div class = "row text-center">
   <div class = "col-xs-12 well message">
     The message will go here
   </div>
 </div>
 <div class = "row text-center">
   <div class = "col-xs-12">
     <button id = "getMessage" class = "btn btn-primary">
       Get Message
     </button>
   </div>
 </div>
</div>

```

### PR DESCRIPTION
**Pull request by arialaurel**
- Fixes Convert JSON to HTML lesson (Closes #4016)
- Add Formatting to Geolocation lesson output (Closes #4010)
- Fixes Render Images lesson
- Removes unnecessary `?callback=` from JSON challenges. (Closes #3972)
- Lots of other formatting and consistency changes.
- Also closes #3962 and closes #3969

### CODE REVIEW

**Comment by bugron** [comment]
@ltegman I've caused conflicts by merging https://github.com/FreeCodeCamp/FreeCodeCamp/pull/3986 which solves the second point from your list, so, please, `rebase` it.

**Comment by ltegman** [comment]
Resolved :smile:

File: seed/challenges/json-apis-and-ajax.json [hunk 1]
```diff
@@ -102,55 +101,55 @@
...
         "assert(editor.match(/\\s*?\\}\\s*?\\)\\s*?\\;/gi) && editor.match(/\\,\\s*?function\\s*?\\(\\s*?\\w*?\\s*?\\)\\s*?\\{/gi) && editor.match(/\\s*?\\}\\s*?\\)\\s*?\\;/gi).length === editor.match(/\\s*?function\\s*?\\(\\s*?\\w*?\\s*?\\)\\s*?\\{/gi).length, 'Each callback function should have a closing set of brackets and parenthesis.')",
-        "assert(editor.match(/\\$\\s*?\\.\\s*?getJSON\\s*?\\(\\s*?\"\\\/json\\\/cats\\.json\\?callback\\=\"\\s*?\\,\\s*?function\\s*?\\(\\s*?json\\s*?\\)\\s*?\\{/gi), 'You should be making use of the getJSON method given in the description to load data from the json file.')",
-        "assert(editor.match(/\\$\\s*?\\(\\s*?\\\"\\.message\\\"\\s*?\\)\\s*?\\.\\s*?html\\s*?\\(\\s*?JSON\\s*?\\.\\s*?stringify\\s*?\\(\\s*?json\\s*?\\)\\s*?\\)/gi), 'Don\\'t forget to make the <code>.html</code> change the contents of the message box so that it contains the result of the getJSON.')"
+        "assert(editor.match(/\\$\\s*?\\.\\s*?getJSON\\s*?\\(\\s*?(\\\"|\\')\\\/json\\\/cats\\.json\"\\s*?\\,\\s*?function\\s*?\\(\\s*?json\\s*?\\)\\s*?\\{/gi), 'You should be making use of the getJSON method given in the description to load data from the json file.')",
```
Comment (BerkeleyTrue) [inline seed/challenges/json-apis-and-ajax.json]: This test no longer passes

Comment (ltegman) [inline seed/challenges/json-apis-and-ajax.json]: Good catch, I knew I'd miss something. Working on it now and will update the PR soon.

**Comment by ltegman** [comment]
Leftover regex `\"` replaced with `(\\\"|\\')` as per @BerkeleyTrue's comment.

**Comment by BerkeleyTrue** [comment]
:+1:
